--- origsrc/libopensync-0.22/formats/Makefile.am	2007-03-27 06:49:27.000000000 -0500
+++ src/libopensync-0.22/formats/Makefile.am	2008-05-09 16:10:49.968750000 -0500
@@ -12,27 +12,27 @@
 formats_LTLIBRARIES = data.la event.la todo.la contact.la note.la file.la
 
 data_la_SOURCES = data.c
-data_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-data_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+data_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+data_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 event_la_SOURCES = event.c
-event_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-event_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+event_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+event_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 note_la_SOURCES = note.c
-note_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-note_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+note_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+note_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 todo_la_SOURCES = todo.c
-todo_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-todo_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+todo_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+todo_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 contact_la_SOURCES = contact.c
-contact_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-contact_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+contact_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+contact_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 file_la_SOURCES = file.c
-file_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-file_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+file_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+file_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 SUBDIRS = vformats-xml
--- origsrc/libopensync-0.22/formats/vformats-xml/Makefile.am	2007-03-27 06:49:27.000000000 -0500
+++ src/libopensync-0.22/formats/vformats-xml/Makefile.am	2008-05-09 16:10:49.984375000 -0500
@@ -22,29 +22,29 @@
 lib_LTLIBRARIES =  libopensync-xml.la
 
 xml_vcard_la_SOURCES = xml-vcard.c vformat.c
-xml_vcard_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-xml_vcard_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
+xml_vcard_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+xml_vcard_la_LIBADD = libopensync-xml.la $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
 
 xml_vcal_la_SOURCES = xml-vcal.c vformat.c vcalical.c
-xml_vcal_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-xml_vcal_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
+xml_vcal_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+xml_vcal_la_LIBADD = libopensync-xml.la $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
 
 xml_vnote_la_SOURCES = xml-vnote.c vformat.c
-xml_vnote_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-xml_vnote_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
+xml_vnote_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+xml_vnote_la_LIBADD = libopensync-xml.la $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
 
 xml_evolution_la_SOURCES = xml-evolution.c vformat.c
-xml_evolution_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-xml_evolution_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
+xml_evolution_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+xml_evolution_la_LIBADD = libopensync-xml.la $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
 
 xml_kde_la_SOURCES = xml-kde.c vformat.c
-xml_kde_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-xml_kde_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
+xml_kde_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+xml_kde_la_LIBADD = libopensync-xml.la $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
 
 xmldoc_la_SOURCES = xmldoc.c
-xmldoc_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-xmldoc_la_LIBADD = @PACKAGE_LIBS@ @LIBS@ libopensync-xml.la
+xmldoc_la_LDFLAGS = -module -avoid-version -export-dynamic -no-undefined @GCOV_LDFLAGS@
+xmldoc_la_LIBADD = libopensync-xml.la $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
 
 libopensync_xml_la_SOURCES = xml-support.c
-libopensync_xml_la_LDFLAGS = @GCOV_LDFLAGS@
-libopensync_xml_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+libopensync_xml_la_LDFLAGS = @GCOV_LDFLAGS@ -no-undefined
+libopensync_xml_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @XML_LIBS@ @LIBS@
--- origsrc/libopensync-0.22/opensync/Makefile.am	2007-03-27 06:49:18.000000000 -0500
+++ src/libopensync-0.22/opensync/Makefile.am	2008-05-09 16:10:50.000000000 -0500
@@ -77,5 +77,5 @@
 	*.bbg \
 	*.da
 
-libopensync_la_LDFLAGS = @PACKAGE_LIBS@ @GCOV_LDFLAGS@
+libopensync_la_LDFLAGS = @PACKAGE_LIBS@ @GCOV_LDFLAGS@ -no-undefined
 libopensync_la_LIBADD = @XML_LIBS@
--- origsrc/libopensync-0.22/osengine/Makefile.am	2007-03-27 06:49:26.000000000 -0500
+++ src/libopensync-0.22/osengine/Makefile.am	2008-05-09 16:10:50.031250000 -0500
@@ -43,5 +43,5 @@
 	*.bbg \
 	*.da
 
-libosengine_la_LDFLAGS = $(PACKAGE_LIBS) -R $(libdir) @GCOV_LDFLAGS@
+libosengine_la_LDFLAGS = $(PACKAGE_LIBS) -R $(libdir) @GCOV_LDFLAGS@ -no-undefined
 libosengine_la_LIBADD = $(top_builddir)/opensync/libopensync.la
--- origsrc/libopensync-0.22/tests/mock-plugin/Makefile.am	2007-03-27 06:49:19.000000000 -0500
+++ src/libopensync-0.22/tests/mock-plugin/Makefile.am	2008-05-09 16:14:00.828125000 -0500
@@ -14,9 +14,9 @@
 endif
 
 mock_sync_la_SOURCES = mock_sync.c
-mock_sync_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-mock_sync_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+mock_sync_la_LDFLAGS = -avoid-version -export-dynamic -module -no-undefined @GCOV_LDFLAGS@
+mock_sync_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
 
 mockformat_la_SOURCES = mock_format.c
-mockformat_la_LDFLAGS = -avoid-version -export-dynamic -module @GCOV_LDFLAGS@
-mockformat_la_LIBADD = @PACKAGE_LIBS@ @LIBS@
+mockformat_la_LDFLAGS = -avoid-version -export-dynamic -module -no-undefined @GCOV_LDFLAGS@
+mockformat_la_LIBADD = $(top_builddir)/opensync/libopensync.la @PACKAGE_LIBS@ @LIBS@
--- origsrc/libopensync-0.22/wrapper/Makefile.am	2007-03-27 06:49:09.000000000 -0500
+++ src/libopensync-0.22/wrapper/Makefile.am	2008-05-09 16:10:50.500000000 -0500
@@ -22,5 +22,5 @@
 endif
 
 nodist__opensync_la_SOURCES = opensync_wrap.c
-_opensync_la_LDFLAGS = @PACKAGE_LIBS@ @XML_LIBS@ $(PYTHON_LDFLAGS) -R $(libdir) -module -avoid-version
+_opensync_la_LDFLAGS = @PACKAGE_LIBS@ @XML_LIBS@ $(PYTHON_LDFLAGS) -R $(libdir) -module -avoid-version -no-undefined
 _opensync_la_LIBADD = $(top_builddir)/opensync/libopensync.la
