--- origsrc/libopensync-0.36/cmake/OpenSyncDefaults.cmake	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/cmake/OpenSyncDefaults.cmake	2008-05-09 16:18:50.015625000 -0500
@@ -26,7 +26,7 @@
 
 # Installation directories:
 
-SET( OPENSYNC_API_DIR "opensync-1.0" )
+SET( OPENSYNC_API_DIR "opensync-0.36" )
 SET( OPENSYNC_PLUGINDIR "${LIB_INSTALL_DIR}/${OPENSYNC_API_DIR}/plugins" CACHE PATH "OpenSync plugin directory" )
 SET( OPENSYNC_FORMATSDIR "${LIB_INSTALL_DIR}/${OPENSYNC_API_DIR}/formats" CACHE PATH "OpenSync format plugin directory" )
 SET( OPENSYNC_PYTHON_PLUGINDIR "${LIB_INSTALL_DIR}/${OPENSYNC_API_DIR}/python-plugins" CACHE PATH "OpenSync python plugin directory" )
--- origsrc/libopensync-0.36/opensync/format/opensync_format_env.c	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/opensync/format/opensync_format_env.c	2008-05-09 16:18:50.046875000 -0500
@@ -71,7 +71,7 @@
 	while ((de = g_dir_read_name(dir))) {
 		filename = g_strdup_printf ("%s/%s", path, de);
 		
-		if (!g_file_test(filename, G_FILE_TEST_IS_REGULAR) || !g_pattern_match_simple("*.so", filename)) {
+		if (!g_file_test(filename, G_FILE_TEST_IS_REGULAR) || !g_pattern_match_simple("*."G_MODULE_SUFFIX, filename)) {
 			g_free(filename);
 			continue;
 		}
--- origsrc/libopensync-0.36/opensync/plugin/opensync_plugin_env.c	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/opensync/plugin/opensync_plugin_env.c	2008-05-09 16:18:50.078125000 -0500
@@ -147,7 +147,7 @@
 	while ((de = g_dir_read_name(dir))) {
 		filename = g_strdup_printf ("%s/%s", path, de);
 		
-		if (!g_file_test(filename, G_FILE_TEST_IS_REGULAR) || !g_pattern_match_simple("*.so", filename)) {
+		if (!g_file_test(filename, G_FILE_TEST_IS_REGULAR) || !g_pattern_match_simple("*."G_MODULE_SUFFIX, filename)) {
 			g_free(filename);
 			continue;
 		}
--- origsrc/libopensync-0.36/tests/format-tests/check_format_env.c	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/tests/format-tests/check_format_env.c	2008-05-09 16:20:20.031250000 -0500
@@ -369,7 +369,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/mock-format.so", curdir);
+	char *path = g_strdup_printf("%s/mock-format.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(osync_module_load(module, path, &error), NULL);
 	fail_unless(error == NULL, NULL);
 	g_free(path);
--- origsrc/libopensync-0.36/tests/module-tests/check_module.c	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/tests/module-tests/check_module.c	2008-05-09 16:22:40.859375000 -0500
@@ -27,7 +27,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/mock-format.so", curdir);
+	char *path = g_strdup_printf("%s/mock-format.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(osync_module_load(module, path, &error), NULL);
 	fail_unless(error == NULL, NULL);
 	g_free(path);
@@ -51,7 +51,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/does-not-exist.so", curdir);
+	char *path = g_strdup_printf("%s/does-not-exist.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(!osync_module_load(module, path, &error), NULL);
 	fail_unless(error != NULL, NULL);
 	g_free(path);
@@ -74,7 +74,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/mock-format.so", curdir);
+	char *path = g_strdup_printf("%s/mock-format.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(osync_module_load(module, path, &error), NULL);
 	fail_unless(error == NULL, NULL);
 	g_free(path);
@@ -102,7 +102,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/mock-format.so", curdir);
+	char *path = g_strdup_printf("%s/mock-format.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(osync_module_load(module, path, &error), NULL);
 	fail_unless(error == NULL, NULL);
 	g_free(path);
@@ -131,7 +131,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/mock-format.so", curdir);
+	char *path = g_strdup_printf("%s/mock-format.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(osync_module_load(module, path, &error), NULL);
 	fail_unless(error == NULL, NULL);
 	g_free(path);
@@ -158,7 +158,7 @@
 	fail_unless(error == NULL, NULL);
 	
 	char *curdir = g_get_current_dir();
-	char *path = g_strdup_printf("%s/mock-format.so", curdir);
+	char *path = g_strdup_printf("%s/mock-format.%s", curdir, G_MODULE_SUFFIX);
 	fail_unless(osync_module_load(module, path, &error), NULL);
 	fail_unless(error == NULL, NULL);
 	g_free(path);
--- origsrc/libopensync-0.36/tests/support.c	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/tests/support.c	2008-05-09 16:21:21.062500000 -0500
@@ -48,17 +48,17 @@
 		abort();
 	g_free(command);*/
 	
-	command = g_strdup_printf("cp ./mock-plugin/mock-sync.so %s", testbed);
+	command = g_strdup_printf("cp ./mock-plugin/mock-sync.%s %s", G_MODULE_SUFFIX, testbed);
 	if (system(command))
 		abort();
 	g_free(command);
 	
-	command = g_strdup_printf("cp ./mock-plugin/mock-format.so %s", testbed);
+	command = g_strdup_printf("cp ./mock-plugin/mock-format.%s %s", G_MODULE_SUFFIX, testbed);
 	if (system(command))
 		abort();
 	g_free(command);
 	
-	command = g_strdup_printf("cp -R ../formats/*.so %s", testbed);
+	command = g_strdup_printf("cp -R ../formats/*.%s %s", G_MODULE_SUFFIX, testbed);
 	if (system(command))
 		abort();
 	g_free(command);
--- origsrc/libopensync-0.36/wrapper/CMakeLists.txt	2008-01-26 11:01:34.000000000 -0600
+++ src/libopensync-0.36/wrapper/CMakeLists.txt	2008-05-09 16:18:50.125000000 -0500
@@ -25,7 +25,7 @@
 	   COMMENT "Munging SWIG-generated Python class"
 	   VERBATIM)
 
-	INSTALL( FILES ${CMAKE_CURRENT_BINARY_DIR}/_opensync.so
+	INSTALL( FILES ${CMAKE_CURRENT_BINARY_DIR}/_opensync${CMAKE_SHARED_MODULE_SUFFIX}
 	   DESTINATION ${LIB_INSTALL_DIR}/python${PYTHON_VERSION}/site-packages/ )
 	INSTALL( FILES ${CMAKE_CURRENT_BINARY_DIR}/opensync_hack.py
 	   DESTINATION ${LIB_INSTALL_DIR}/python${PYTHON_VERSION}/site-packages/ RENAME opensync.py )
